### PR TITLE
Fix warning level calculation for .NET 10

### LINKED ISSUE #76226: [.NET 10] certain `[Obsolete]` warnings are not reported
**Version Used**:  .NET SDK 10.0.100-alpha.1.24573.1

We noticed this from Maestro/codeflow here:

* https://github.com/dotnet/android/pull/9395

**Steps to Reproduce**:

1. Call `AppDomain.CreateDomain("test")` in .NET 10
2. Note that you don't get a `SYSLIB0024` warning

Repro: [testroslyn.zip](https://github.com/user-attachments/files/17981183/testroslyn.zip)

Note that this seems to work fine using `net9.0` and a .NET 9 SDK.

**Diagnostic Id**: 

```
SYSLIB0024: 'AppDomain.CreateDomain(string)' is obsolete: 'Creating and unloading AppDomains is not supported and throws an exception.'
```

**Expected Behavior**:

I get a `SYSLIB0024` warning.

**Actual Behavior**:

I get no `SYSLIB0024` warning.

**Other info**:

Note that if I open `packs\Microsoft.NETCore.App.Ref\10.0.0-alpha.1.24570.9\ref\net10.0\System.Runtime.dll`, it looks correct:

Image

The warning also seems to work fine if I declare an `[Obsolete]` method within the C# file.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/roslyn/issues/76226

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets [hunk 1]
```diff
@@ -62,7 +62,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- The CSharp.props used to hard-code WarningLevel to 4, so to maintain parity with .NET Framework projects we do that here. -->
     <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == '' And '$(TargetFrameworkIdentifier)' == '.NETFramework' ">4</WarningLevel>
     <!-- .NET projects, however, can float up to their TFM's major version -->
-    <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == '' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' ">$(_TargetFrameworkVersionWithoutV.Substring(0, 1))</WarningLevel>
+    <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == '' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' ">$([System.Version]::Parse('$(_TargetFrameworkVersionWithoutV)').Major</WarningLevel>
```
Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets:65]: NOT TESTED YET

Comment (baronfel) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets:65]: Mind taking a look at https://github.com/dotnet/sdk/blob/main/test%2FMicrosoft.NET.Build.Tests%2FGivenThatWeWantToFloatWarningLevels.cs and seeing if there's a place to add a regression test?

Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets]: Manually tested now, and updated the existing test as well. Let's see if all will be green.

**Review by baronfel — APPROVED** [review]

**Comment by baronfel** [comment]
Cc @marcpopMSFT we might have a few more time bombs lying in wait after the net10 transition

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**